pkg/skaffold/docker/image_test.go

```diff
@@ -316,3 +319,70 @@ func TestRepoDigest(t *testing.T) {
 		})
 	}
 }
+
+func TestInsecureRegistry(t *testing.T) {
+	called := false // variable to make sure we've called our getInsecureRegistry function
+	getInsecureRegistryImpl = func(_ string) (name.Reference, error) {
+		called = true
+		return name.Tag{}, nil
+	}
+	getRemoteImageImpl = func(_ name.Reference) (v1.Image, error) {
+		return random.Image(0, 0)
+	}
+
+	tests := []struct {
+		name               string
+		image              string
+		insecureRegistries map[string]bool
+		insecure           bool
+		shouldErr          bool
+	}{
+		{
+			name:               "secure image",
+			image:              "gcr.io/secure/image",
+			insecureRegistries: map[string]bool{},
+		},
+		{
+			name:  "insecure image",
+			image: "my.insecure.registry/image",
+			insecureRegistries: map[string]bool{
+				"my.insecure.registry": true,
+			},
+			insecure: true,
+		},
+		{
+			name:      "insecure image not provided by user",
+			image:     "my.insecure.registry/image",
+			insecure:  true,
+			shouldErr: true,
+		},
+		{
+			name:  "secure image provided in insecure registries list",
```

Comment:
I think this case is testing if user incorrectly adds a secure registry to insecure list by using a hint "test.insecure".
Right now, we do not validate the `insecureRegistry` list hence it is doing what its supposed to do.  
Shd we not test this? or add a todo to ring this test back in when we have some validation?

Reply:
not sure I understand what you're saying here. if you're asking about validation of the insecure registry list (as in, each specified registry is actually "insecure"), it's not really possible to validate this without pinging each individual registry before making the `remote.Image` call. if any registries are passed by the user as "insecure", the list will be populated and `getInsecureRegistryImpl` will be called (which is part of what this test is testing). does that answer your question?